meson.build

```diff
@@ -776,6 +777,40 @@ configure_file(
 
 add_project_arguments('-include', meson.current_build_dir() / 'config.h', language : 'c')
 
+compiler_flags = [
+  '-fno-common',
+  '-Wall',
+  '-Wextra',
```

Comment:
Ironically for a PR that "enables warning flags" and fixes issues, this actually adds warnings produced by Meson itself:
```
Compiler for C supports arguments -Wall: YES 
meson.build:810: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
Compiler for C supports arguments -Wextra: YES 
meson.build:810: WARNING: Consider using the built-in warning_level option instead of using "-Wextra".
```